### PR TITLE
*: fix some error related issues

### PR DESCRIPTION
**Pull request by breeswish**
Fix:
- Some errors do not have correct parameters.
- Errors that contain "%" in the message body are not correctly printed to the client.

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM
Please make sure tidb-test can pass it, some of them check error message.

File: terror/terror.go [hunk 1]
```diff
@@ -257,7 +257,7 @@ func (e *Error) NotEqual(err error) bool {
 // ToSQLError convert Error to mysql.SQLError.
 func (e *Error) ToSQLError() *mysql.SQLError {
 	code := e.getMySQLErrorCode()
-	return mysql.NewErrf(code, e.getMsg())
+	return mysql.NewErrf(code, "%s", e.getMsg())
```
Comment (shenli) [inline terror/terror.go]: Can we use mysql.NewErr() here?

**Comment by shenli** [comment]
@breeswish Please resolve the conflicts.

File: terror/terror.go (hunk 1 above)
Comment (breeswish) [inline terror/terror.go]: @shenli `mysql.NewErr` cannot be used here. It's functionality is not what we usually expected to be:

mysql.NewErr: NewErr generates a SQL error, with an error code and default format specifier defined in MySQLErrName.

```go
func NewErr(errCode uint16, args ...interface{}) *SQLError
```

mysql.NewErrf: NewErrf creates a SQL error, with an error code and a format specifier.

```go
func NewErrf(errCode uint16, format string, args ...interface{}) *SQLError
```

**Comment by breeswish** [comment]
@shenli Done

**Comment by shenli** [comment]
LGTM

**Pull request merged**